src/pytorch_lightning/callbacks/quantization.py

```diff
@@ -181,7 +184,9 @@ def __init__(
             )
         self._observer_type = observer_type
 
-        if collect_quantization is not None and not isinstance(collect_quantization, (int, Callable)):
+        if collect_quantization is not None and not (
+            isinstance(collect_quantization, int) and callable(collect_quantization)
+        ):
```

Comment:
Uh oh, @krishnakalyan3, I just noticed several failing tests, that are linked to the changes made here. Would you mind taking a look at them?

```suggestion
        if collect_quantization is not None and not (
            isinstance(collect_quantization, int) or callable(collect_quantization)
        ):
```

Reply:
`isinstance` check against a tuple of types is equivalent to `or`, not `and`

Reply:
checking...